### DISCUSSION TITLE
connection errored out

### DISCUSSION THREAD

**Discussion by sannudaboss** [lllyasviel/Fooocus#1925] [Q&A]
i just cant find answer to my problem soo maybe someone here can help. i downloaded everything like i should and if i run fooocus, it takes few seconds and then just error, and says connection errored out, i know that system swap something might be the problem, but i tried to fix that with the tutorial that they show, but it was already correct. Soo im bit lost here and dont know what to do. Sorry for bad english, if you dont understand what im saying, please ask and i will try to explain better. but i really need help.

I will also add this, this is after first download and run. ive tried reinstalling and same error.
C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Fast-forward merge
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.862
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_version6Rundiffusion.safetensors" to C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 6.62G/6.62G [21:37<00:00, 5.48MB/s]
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 6144 MB, total RAM 8046 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce GTX 1660 SUPER : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.

C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831>pause
Press any key to continue . . .

**Comment by eddyizm** [comment]
I don't see where you are getting the connection time out message?

**Reply by sannudaboss** [reply]
it doesnt say that in there but only in the UI or whats that called, and its like error in a red box with text below that say connection errored out, not time out

**Reply by mashb1t** [reply]
please check if you can run Fooocus with `--always-cpu` or if this also causes errors.

**Reply by sannudaboss** [reply]
you mean in colab? I dont know where can i put that command if not in colab and id prefer not to use colab (tho im using it right now to make as much as possible) cuz the faceswapping always crashes it with the same error. but as far as i know if i can get fooocus working from my computer, its completely free and shouldnt have any problems with faceswaping? but speaking about the command, i just thought about it, do you mean i edit the run file and add the command there? i thought i had amd gpu soo i already edited the run file for amd and now the launch code is bit different. imma put that here too. basically if yall can help me out i am super thankful for that and i know my english and being a noob beginner is super annoying but im trying to be better so i hope yall dont give up :) if yall could help me out to get no errors maybe in discord call or something, i could pay you also :) oh and i remember in one video i watched on installing fooocus, he had 2 things downloaded on a first run, but i have only 1 soo yeah. 

this is now when i have added the amd gpu thingy :) not sure if that was even needed. i tried generating 1 picture but it made my computer supper laggy right now, it didnt happen before, only now when i added the amd gpu thing. 

C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831>.\python_embeded\python.exe -m pip uninstall torch torchvision torchaudio torchtext functorch xformers -y
Found existing installation: torch 2.0.0
Uninstalling torch-2.0.0:
  Successfully uninstalled torch-2.0.0
Found existing installation: torchvision 0.15.1
Uninstalling torchvision-0.15.1:
  Successfully uninstalled torchvision-0.15.1
WARNING: Skipping torchaudio as it is not installed.
WARNING: Skipping torchtext as it is not installed.
WARNING: Skipping functorch as it is not installed.
WARNING: Skipping xformers as it is not installed.

C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831>.\python_embeded\python.exe -m pip install torch-directml
Requirement already satisfied: torch-directml in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (0.2.0.dev230426)
Collecting torch==2.0.0 (from torch-directml)
  Using cached torch-2.0.0-cp310-cp310-win_amd64.whl (172.3 MB)
Collecting torchvision==0.15.1 (from torch-directml)
  Using cached torchvision-0.15.1-cp310-cp310-win_amd64.whl (1.2 MB)
Requirement already satisfied: filelock in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torch==2.0.0->torch-directml) (3.12.2)
Requirement already satisfied: typing-extensions in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torch==2.0.0->torch-directml) (4.7.1)
Requirement already satisfied: sympy in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torch==2.0.0->torch-directml) (1.12)
Requirement already satisfied: networkx in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torch==2.0.0->torch-directml) (3.1)
Requirement already satisfied: jinja2 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torch==2.0.0->torch-directml) (3.1.2)
Requirement already satisfied: numpy in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torchvision==0.15.1->torch-directml) (1.23.5)
Requirement already satisfied: requests in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torchvision==0.15.1->torch-directml) (2.31.0)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from torchvision==0.15.1->torch-directml) (9.2.0)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from jinja2->torch==2.0.0->torch-directml) (2.1.3)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from requests->torchvision==0.15.1->torch-directml) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from requests->torchvision==0.15.1->torch-directml) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from requests->torchvision==0.15.1->torch-directml) (2.0.3)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from requests->torchvision==0.15.1->torch-directml) (2023.5.7)
Requirement already satisfied: mpmath>=0.19 in c:\users\admin\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from sympy->torch==2.0.0->torch-directml) (1.3.0)
DEPRECATION: torchsde 0.2.5 has a non-standard dependency specifier numpy>=1.19.*; python_version >= "3.7". pip 23.3 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of torchsde or contact the author to suggest that they release a version with a conforming dependency specifiers. Discussion can be found at https://github.com/pypa/pip/issues/12063
Installing collected packages: torch, torchvision
  WARNING: The scripts convert-caffe2-to-onnx.exe, convert-onnx-to-caffe2.exe and torchrun.exe are installed in 'C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed torch-2.0.0 torchvision-0.15.1

[notice] A new release of pip is available: 23.2.1 -> 23.3.2
[notice] To update, run: C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\python.exe -m pip install --upgrade pip

C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --directml
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--directml']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.862
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Using directml with device:
Total VRAM 1024 MB, total RAM 8046 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: privateuseone
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using split attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using split attention in VAE
extra {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
Base model loaded: C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors].
Loaded LoRA [C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 8297779955243327997
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] boy, intricate, highly detailed, flowing, dramatic light, sharp focus, cinematic, fine detail, elegant, sublime, vibrant colors, epic, joyful, magical atmosphere, professional, winning, unique, iconic, creative, beautiful, awesome, surreal, artistic, best, perfect,, colorful, illuminated, gorgeous, striking, pretty, amazing, pure, very inspirational, cool
[Fooocus] Preparing Fooocus text #2 ...
[Prompt Expansion] boy, highly detailed, professional fine detail, polished, complex, exquisite, amazing composition, cinematic, elegant, rich colors, beautiful scenic background, inspired, great focus, creative, wonderful atmosphere, positive, unique, cute, adorable, best, imagined, iconic, gorgeous, dramatic, perfect, intricate, artistic, awesome, sharp, colorful, light, color, epic
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding positive #2 ...
[Fooocus] Encoding negative #1 ...
[Fooocus] Encoding negative #2 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (896, 1152)
Preparation time: 19.71 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
Keyboard interruption in main thread... closing server.
ERROR diffusion_model.output_blocks.1.1.transformer_blocks.9.ff.net.0.proj.weight [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 52428800 bytes.
ERROR diffusion_model.output_blocks.1.1.transformer_blocks.9.ff.net.2.weight [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 26214400 bytes.
ERROR diffusion_model.output_blocks.1.1.transformer_blocks.9.attn2.to_k.weight [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 10485760 bytes.
Traceback (most recent call last):
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\async_worker.py", line 823, in worker
    handler(task)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\async_worker.py", line 754, in handler
    imgs = pipeline.process_diffusion(
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\default_pipeline.py", line 361, in process_diffusion
    sampled_latent = core.ksampler(
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\core.py", line 313, in ksampler
    samples = ldm_patched.modules.sample.sample(model,
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\sample.py", line 94, in sample
    real_model, positive_copy, negative_copy, noise_mask, models = prepare_sampling(model, noise.shape, positive, negative, noise_mask)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\sample.py", line 87, in prepare_sampling
    ldm_patched.modules.model_management.load_models_gpu([model] + models, model.memory_required([noise_shape[0] * 2] + list(noise_shape[1:])) + inference_memory)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\patch.py", line 441, in patched_load_models_gpu
    y = ldm_patched.modules.model_management.load_models_gpu_origin(*args, **kwargs)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\model_management.py", line 434, in load_models_gpu
    cur_loaded_model = loaded_model.model_load(lowvram_model_memory)
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\model_management.py", line 301, in model_load
    raise e
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\model_management.py", line 297, in model_load
    self.real_model = self.model.patch_model(device_to=patch_model_to) #TODO: do something with loras and offloading to CPU
  File "C:\Users\ADMIN\Desktop\Fooocus_win64_2-1-831\Fooocus\ldm_patched\modules\model_patcher.py", line 200, in patch_model
    temp_weight = weight.to(torch.float32, copy=True)
RuntimeError: [enforce fail at ..\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 10485760 bytes.

**Reply by mashb1t** [reply]
First of all, you do not have to run the pip commands for directml every time you start Fooocus.

Second, it seems like you have insufficient VRAM (only 1GB recognized, make sure to allocate at least 8GB) / RAM and it can't outsource to swap (might not be configured correctly). Make tsure to check both.
Your previous terminal log shows the usage of a 1660 Super, todays shows an AMD GPU. Did you switch your systems/GPU?

**Reply by sannudaboss** [reply]
yeah i edited the run file to amd gpu. its actually 1660 super yes soo i should change that back to normal? and can you help me how can i do that part ---> (Second, it seems like you have insufficient VRAM (only 1GB recognized, make sure to allocate at least 8GB) / RAM and it can't outsource to swap (might not be configured correctly). Make tsure to check both.)
and the pip commands it runs itself automatically idk

**Reply by mashb1t** [reply]
Why would you adjust the entrypoint for AMD when using an Nvidia card? O.o
Please revert your changes and double check if you have swap set up correctly.

**Comment by sannudaboss** [comment]
okay i reinstalled fooocus, and i think my swap is set up correctly, i will put the screenshot here.
![image_2024-01-16_123053711](https://github.com/lllyasviel/Fooocus/assets/156262895/7eb1607c-82ed-4ddb-9775-714fcb1a61df)

and when it launched, when i tried to do just click on input image box, it automatically crashed again, and when it crashes the command says pause and press any key to continue. i will also add a screenshot of the fooocus tab where it shows the error.
![image_2024-01-16_123422242](https://github.com/lllyasviel/Fooocus/assets/156262895/88a34239-3b53-45e1-9479-39d0848c5c79)



C:\Users\ADMIN\Downloads\Fooocus_win64_2-1-831>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Fast-forward merge
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.862
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_version6Rundiffusion.safetensors" to C:\Users\ADMIN\Downloads\Fooocus_win64_2-1-831\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 6.62G/6.62G [14:28<00:00, 8.18MB/s]
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 6144 MB, total RAM 8046 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce GTX 1660 SUPER : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE

C:\Users\ADMIN\Downloads\Fooocus_win64_2-1-831>pause
Press any key to continue . . .

**Reply by mashb1t** [reply]
@sannudaboss Do you have sufficient disk space to support RAM overflow to swap / disk?(~40GB of free space on each disk)

**Reply by sannudaboss** [reply]
i have only 1 disk and right now 155GB free
